### PR TITLE
CI Check that a chart version was incremented

### PR DESCRIPTION
**Pull request by mattfarina**
This commit does a couple things
1. It reformats the layout to have checks via self contained
   functions rather than all in one loop.
2  Adds a check that the chart.version was incremented

Ref #2373 

For an example failure on version bump see https://circleci.com/gh/mattfarina/charts/52
For an example pass on version bump see https://circleci.com/gh/mattfarina/charts/51

/cc @viglesiasce @prydonius

### CODE REVIEW

**Comment by mattfarina** [comment]
Wonder why it didn't check for the cla this time.

**Comment by mattfarina** [comment]
Note, the issues that were part of my previous comment were due to a GitHub issue with replication of dbs and things showing up in the UI. It seems to be working now.

File: test/circle/lint.sh [hunk 1]
```diff
@@ -30,6 +30,67 @@ run() {
...
+semvercompare() {
+  printf "\nChecking the Chart version has increased for the chart at ${1}\n"
+
+  # Checkout the Chart.yaml file on master to read the version for comparison
+  git show k8s/master:$1/Chart.yaml > /tmp/Chart.yaml
+
+  local oldVer=`yaml r /tmp/Chart.yaml version`
+  local newVer=`yaml r $1/Chart.yaml version`
+
+  # Pre-releases may not be API compatible. So, when tools compare versions
```
Comment (prydonius) [inline test/circle/lint.sh:66]: We should probably review what charts are using pre-releases and move them away from it. There shouldn't be any charts in stable using pre-release versions and those are using it to denote the version of the packaged app should adopt appVersion instead.

**Review by prydonius — APPROVED** [review]
/lgtm

**Pull request merged**